### PR TITLE
Don't run binaryen postprocessing for Emscripten EH/SjLj

### PR DESCRIPTION
**Pull request by aheejin**
Now that we are renaming invoke wrappers and `emscripten_longjmp_jmpbuf`
in the wasm backend, using Emscripten EH or SjLj does not need
Binaryen's postprocessing. This makes Emscripten not enable Binaryen
postprocessing using wasm-emscripten-finalize when we are using them,
and deletes all special handling for `emscripten_longjmp_jmpbuf`, given
that Emscripten will not see it.

Addresses:
WebAssembly/binaryen#3043
WebAssembly/binaryen#3081

Companions:
https://reviews.llvm.org/D88697
WebAssembly/binaryen#3191

### CODE REVIEW

**Comment by aheejin** [comment]
The CI is not gonna pass until we land these companion LLVM and Binaryen patches:
https://reviews.llvm.org/D88697
WebAssembly/binaryen#3191

File: src/deps_info.json [hunk 1]
```diff
@@ -123,7 +123,7 @@
   "setjmp": ["setThrew", "realloc", "testSetjmp", "saveSetjmp"],
   "longjmp": ["setThrew", "realloc", "testSetjmp", "saveSetjmp"],
   "siglongjmp": ["setThrew", "realloc", "testSetjmp", "saveSetjmp"],
-  "emscripten_longjmp_jmpbuf": ["setThrew", "realloc", "testSetjmp", "saveSetjmp"],
+  "emscripten_longjmp": ["setThrew", "realloc", "testSetjmp", "saveSetjmp"],
```
Comment (kripken) [inline src/deps_info.json:126]: to be sure I understand, now the LLVM backend will never emit `emscripten_longjmp_jmpbuf` any more, but it does emit `emscripten_longjmp`?

File: tests/test_other.py [hunk 2]
```diff
@@ -9045,9 +9044,7 @@ def fail(args, details):
 
     # plain -O0
     legalization_message = 'to disable int64 legalization (which requires changes after link) use -s WASM_BIGINT'
-    longjmp_message = 'to disable longjmp support (which requires changes after link) use -s SUPPORT_LONGJMP=0'
     fail([], legalization_message)
-    fail(['-sWASM_BIGINT'], longjmp_message)
     fail(['-sSUPPORT_LONGJMP=0'], legalization_message)
```
Comment (kripken) [inline tests/test_other.py:9048]: i think this line is not needed any more

**Review by kripken — APPROVED** [review]
Great!

File: tests/test_other.py (hunk 2 above)
Comment (aheejin) [reply tests/test_other.py:9048]: Removed.

File: src/deps_info.json (hunk 1 above)
Comment (aheejin) [reply src/deps_info.json:126]: Yes.

**Comment by sbc100** [comment]
Awesome!

**Pull request merged**